10up/classifai
# Classify content within existing Taxonomy terms

### Description of the Change
The PR introduces a new setting option in content classification (IBM Watson NLU) to classify content within existing taxonomy terms. By default, it will adhere to the current behavior, recommending and adding new terms to the post. When the user chooses "Only classify based on existing terms" in the classification method, it will recommend terms only from the existing terms.

![image](https://github.com/10up/classifai/assets/10613171/bbae2345-7048-404d-bd7e-5062434f6670)


Closes #592 

### How to test the Change
1. Go to `Tools` > `ClassifAI` > `Language Processing` > `IBM Watson`
1. Choose "Only classify based on existing terms" in the classification method.
1. Create/Edit the post and click the "Suggest terms & tags" button from the classifAI sidebar panel.
1. Verify that suggested terms/tags are from the existing terms only.
1. Go to ClassifAI settings, choose "Add recommended terms " in the classification method and verify that it now suggests new terms/tags as well.
1. Go to ClassifAI settings, and verify that the results of terms/tags in the language processing previewer are also aligned with the current classification method setting.

### Changelog Entry
> Added - New setting option in content classification to classify content within existing taxonomy terms


### Credits
Props @jeffpaul @iamdharmesh 


### Checklist:
